async-nats/src/jetstream/consumer/push.rs

```diff
@@ -468,11 +470,14 @@ impl<'a> futures::Stream for Ordered<'a> {
                                         };
 
                                         let info = jetstream_message.info()?;
-                                        if info.stream_sequence != self.stream_sequence + 1 {
+                                        if info.consumer_sequence != self.consumer_sequence + 1
+                                            && info.stream_sequence != self.stream_sequence + 1
+                                        {
                                             self.subscriber = None;
                                             continue;
                                         }
                                         self.stream_sequence = info.stream_sequence;
+                                        self.consumer_sequence = info.consumer_sequence;
```

Comment:
Sequence will reset when a mismatch is detected tho?

```suggestion
                                        self.consumer_sequence = info.consumer_sequence;
```

Reply:
agree. fixed.